galleries/nxdata/plot_image.py

```diff
@@ -0,0 +1,51 @@
+"""
+=====
+Image
+=====
+
+Image on a regular two-dimensional grid.
+
+The x-axis and y-axis define a regular 2D grid. The grid is equally spaced in
+this example but this is not necessarily the case.
+
+.. code::
+
+  @NX_class = "NXroot"
+  @default = "scan1"
+  scan1:
+    @NX_class = "NXentry"
+    @default = "data"
+    data:
+      @NX_class = "NXdata"
+      @axes = ["x", "y"]
+      @signal = "z"
+      x: NX_FLOAT64[16]
+      y: NX_FLOAT64[30]
+      z: NX_FLOAT64[16,30]
+
+Explanation:
+
+1. ``@axes`` has two values which corresponds to the signal rank of two.
+
+2. ``z`` is the default signal to be plotted versus ``x`` and ``y``.
+"""
```

Comment:
3. Row-major storage ordering means image data is indexed as row/column pairs

Reply:
I don't understand what this means.

Reply:
Ok "row-major" means "C-order" or "the last dimension is the fastest dimension".

I'm trying to understand now what "indexed as row/column pairs" means.

Do you mean that `z: NX_FLOAT64[30,16]` refers to the fact that the z matrix has 30 rows and 16 columns? Or is the C-order aspect somehow implied by the definition?

If I think of `z` as a numpy array with `z.shape == (30 16)`, it does not matter whether it is C-order or F-order afaik.

Reply:
Instead of you suggestion (which is good but too complex for me at least) I added

3. ``z`` has 30 rows and 16 columns.

4. ``y`` spans the first dimension of ``z`` and ``x`` the second.

This comment made me also swap the dimensions of hist2d, hist2dmix and fscan2d to be consistent with image: y is the first dimension (i.e. the slow one assuming C-order) and x the second dimension (i.e. the fast on assuming C-order).